hbase-client/src/main/java/org/apache/hadoop/hbase/client/MetricsConnection.java

```diff
@@ -652,7 +652,27 @@ public void updateRpc(MethodDescriptor method, Message param, CallStats stats, T
       concurrentCallsPerServerHist.update(callsPerServer);
     }
     // Update the counter that tracks RPCs by type.
-    final String methodName = method.getService().getName() + "_" + method.getName();
+    String methodName = method.getService().getName() + "_" + method.getName();
+    // Distinguish mutate types.
+    if ("Mutate".equals(method.getName())) {
+      final MutationType type = ((MutateRequest) param).getMutation().getMutateType();
+      switch (type) {
+        case APPEND:
+          methodName += "(Append)";
+          break;
+        case DELETE:
+          methodName += "(Delete)";
+          break;
+        case INCREMENT:
+          methodName += "(Increment)";
+          break;
+        case PUT:
+          methodName += "(Put)";
+          break;
+        default:
+          throw new RuntimeException("Unrecognized mutation type " + type);
```

Comment:
This is for metrics so usually we should not throw exception again, appending `(Unknown)` or just appending nothing is enough?

Reply:
Resolved it.